libr/include/r_util/r_event.h

```diff
@@ -29,12 +38,11 @@ typedef struct r_event_meta_t {
 	const char *string;
 } REventMeta;
 
-typedef void (*REventCallback)(REvent *ev, int type, void *user, void *data);
 
 R_API REvent *r_event_new(void *user);
 R_API void r_event_free(REvent *ev);
-R_API void r_event_hook(REvent *ev, int type, REventCallback cb);
-R_API void r_event_unhook(REvent *ev, int type, REventCallback cb);
+R_API int r_event_hook(REvent *ev, int type, REventCallback cb, void *user);
+R_API void r_event_unhook(REvent *ev, int type, int handle);
```

Comment:
is the type really needed at this point? I think it's enough to use the handle

Reply:
Yeah right I just noticed that too while trying to remove it. I am just not sure whether the caller of the function should be responsible for specifying the handle.